### PR TITLE
Tweak: Kit-Library - Changed back button behavior [ED-3618]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: core/app/modules/kit-library/assets/js/components/layout/header-back-button.js [hunk 1]
```diff
@@ -1,15 +1,20 @@
 import { Button } from '@elementor/app-ui';
+import { useLastFilterContext } from '../../context/last-filter-context';
+import { useNavigate } from '@reach/router';
 
 import './header-back-button.scss';
 
 export default function HeaderBackButton() {
+	const navigate = useNavigate();
+	const { lastFilter } = useLastFilterContext();
```
Comment (StyleShit) [inline core/app/modules/kit-library/assets/js/components/layout/header-back-button.js:9]:
```suggestion
	const navigate = useNavigate(),
		{ lastFilter } = useLastFilterContext();
```

File: core/app/modules/kit-library/assets/js/pages/index/index.js [hunk 2]
```diff
@@ -82,13 +83,16 @@ function useMenuItems( path ) {
  */
 function useRouterQueryParams( queryParams, setQueryParams, exclude = [] ) {
 	const location = useLocation();
+	const { setLastFilter } = useLastFilterContext();
```
Comment (StyleShit) [inline core/app/modules/kit-library/assets/js/pages/index/index.js:86]:
```suggestion
	const location = useLocation(),
		{ setLastFilter } = useLastFilterContext();
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: core/app/modules/kit-library/assets/js/components/layout/header-back-button.js (hunk 1 above)
Comment (Nevoss) [reply core/app/modules/kit-library/assets/js/components/layout/header-back-button.js]: Done

File: core/app/modules/kit-library/assets/js/pages/index/index.js (hunk 2 above)
Comment (Nevoss) [reply core/app/modules/kit-library/assets/js/pages/index/index.js]: Done

**Review by StyleShit — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**